_posts/en/newsletters/2024-01-24-newsletter.md

```diff
@@ -0,0 +1,240 @@
+    chain split.  An attacker could use this to trick the operator of a
+    btcd node (or software connected to a btcd node) into accepting
+    invalid bitcoins.
+
+    The bug was privately disclosed to the btcd maintainers who fixed it
+    in the recent v0.24.0 release.  Anyone using btcd for consensus
+    enforcement is strongly urged to upgrade.  Additionally, Chris
+    Stewart [replied][stewart bitcoin-s] to the Delving Bitcoin thread
+    with a patch for the same failure in the bitcoin-s library.  Authors
+    of any code that might be used to validate BIP68 relative locktimes
+    are urged to check the code for the same flaw.
+
+- **Proposed changes to LN for v3 relay and ephemeral anchors:** Bastien
+  Teinturier [posted][teinturier v3] to Delving Bitcoin to describe the
+  changes he thinks should be made to the LN specification to make
+  effective use of [v3 transaction relay][topic v3 transaction relay]
+  and [ephemeral anchors][topic ephemeral anchors].  The changes appear
+  simple:
+
+  - *Anchor swap:* the commitment transaction's two current [anchor outputs][topic
+    anchor outputs] used to guarantee the ability to [CPFP fee bump][topic cpfp]
+    under the [CPFP carve out][topic cpfp carve out] policy are removed
+    and one ephemeral anchor is added in their place.
+
+  - *Reducing delays:* the extra 1-block delays on the commitment transaction are removed.
+    They were added to ensure that CPFP carve out would always work but are
+    not necessary under the v3 relay policy.
+
+  - *Trimming redirect:* instead of spending the value of all [trimmed HTLCs][topic trimmed
+    htlc] to fees in the commitment transaction, the combined value is
+    added to the value of the anchor output, ensuring the extra fees are
+    used to guarantee confirmation of both the commitment and the ephemeral anchor
+    (see [last week's newsletter][news285 mev] for discussion).
+
+  - *Other changes:* some other minor changes and simplifications are made.
+
+  The subsequent discussion examined several interesting consequences of
+  the proposed changes, including:
+
+  - *Reduced UTXO requirements:* ensuring the correct channel state goes onchain becomes easier
+    thanks to the removal of the extra 1-block delay.  If a faulty party
+    broadcasts a revoked commitment transaction, the other party can use
+    their main output from that commitment to CPFP fee bump that revoked
+    commitment.  They don't need to keep a separate confirmed UTXO for
+    that purpose.  For this to be safe, the party must keep a sufficient
+    reserve balance, as the 1% minimum specified in [BOLT2][] may not be
+    enough to cover fees.  For a non-forwarding node that keeps a
+    sufficient reserve, the only time they need a separate UTXO for
+    security reasons is when they want to accept an incoming payment.
+
+  - *Circular dependencies:* v3 relay depends on [cluster mempool][topic
```

Comment:
v3 doesn't depend on cluster mempool?

Reply:
To elaborate, there isn't a circular dependency problem. The "imbue v3 logic on commitment transactions" idea was proposed by @TheBlueMatt in the LN spec meeting because he thinks it might take longer for LN to switch commitment transaction formats than for Bitcoin Core to merge cluster mempool, and LN development shouldn't block Bitcoin Core development.